### PR TITLE
Homework2

### PR DESCRIPTION
**Pull request by Riko20**

### CODE REVIEW

File: Homework2.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+sort_dict(data)
+
+
+#1.2) сгруппировать данные по значению ключа 'city'

+# вывод должен быть такого вида :
+
+def groupcity(data):
+    if "__main__" == __name__:
```
Comment (mmykhailenko) [inline Homework2.py:28]: no need here

File: Homework2.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    numlist = []
+    for elem in list_var:
+        numlist.append(list_var.count(elem))
+    ind = numlist.index(max(numlist))
+    ourstr = list_var[ind]
+    return ourstr
+
+print ("2:  ", most_frequent(['a', 'a', 'bi', 'bi', 'bi','d','d','d','d','d','d','d']))
+
+# 3) Дано целое число. Необходимо подсчитать произведение всех цифр в этом числе, за исключением нулей.
```
Comment (mmykhailenko) [inline Homework2.py:53]: this task is not completed. Wrong result

File: Homework2.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        return -1
+    res = array[n] ** n
+    print(res)
+
+
+arrayret(array, 6)
+# =======================================================
+# 5) Есть строка со словами и числами, разделенными пробелами (один пробел между словами и/или числами).
+
+def stringint(stringa):
```
Comment (mmykhailenko) [inline Homework2.py:79]: wrong result when - "hello 1 one two three four 15 world"
wrong result when - "hello 1 one two three four five15 world"

File: Homework2.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    counter = 0
+    a = stringa.split()
+    for elem in a:
+        if elem.isalpha():
+            counter += 1
+        if elem.isdecimal():
+            counter =0
+        if counter == 3:
+            print('It`s true')
```
Comment (mmykhailenko) [inline Homework2.py]: better return True or False. Use print only for debug

**Review by mmykhailenko — APPROVED** [review]

File: Homework2.py (hunk 1 above)
Comment (Riko20) [reply Homework2.py:28]: @mmykhailenko, if possible, send please how should to be solved excercise 1.2

File: Homework2.py (hunk 2 above)
Comment (Riko20) [reply Homework2.py:53]: @mmykhailenko, if possible please give me the correct answer, i try to test it with another strings, program returns most common string in the list.

File: Homework2.py (hunk 3 above)
Comment (Riko20) [reply Homework2.py:79]: @mmykhailenko, do u mean that this program must return True or False? As i understand the condition of excercise, if there are 3 string one after another, program must return sequence True or False

Comment (mmykhailenko) [reply Homework2.py:79]: yes 3, but here I have 5 "hello 1 one two three four five15 world"

File: Homework2.py (hunk 2 above)
Comment (mmykhailenko) [reply Homework2.py:53]:
```
res = 1
for d in str(number):
    res *= int(d) if int(d) else 1
return res

```

File: Homework2.py (hunk 1 above)
Comment (mmykhailenko) [reply Homework2.py:28]: if "__main__" == __name__:
I will explain this code on Monday. Please remind me

**Pull request closed without merging**